osu.Game/Rulesets/Objects/Legacy/ConvertHitObjectParser.cs

```diff
@@ -69,30 +75,30 @@ public override HitObject Parse(string text)
             {
                 result = CreateHit(pos, combo, comboOffset);
 
-                if (split.Length > 5)
-                    readCustomSampleBanks(split[5], bankInfo);
+                if (ranges.Length > 5)
```

Comment:
I was using an alternative approach: stackalloc the `Span<Range>` with maximum length we want to support (+1 for guarding), and check the return value of `Split`.
Both approaches should be functionally correct.